lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -761,16 +770,17 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
       return getOrCreateContext(DC->getParent());
     case DeclContextKind::MacroDecl:
       return getOrCreateContext(DC->getParent());
+    case DeclContextKind::ExtensionDecl: {
+      auto *ED = cast<ExtensionDecl>(DC);
+      if (auto *NTD = ED->getExtendedNominal())
```

Comment:
Do you know in what situation an extension wouldn't know what's the nominal it's extending? Is it possible to add that to the NestedTypes test case?

Reply:
According to the doxygen comments this can only happen during type checking when the extended type hasn't been visited yet. I don't think the failure case is reachable during IRGen.